### PR TITLE
Fix holiday dates importer

### PR DESCRIPTION
**Pull request by danigargar**
#### Type Of Change 
- [ ] Small bug fix
- [x] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [x] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [x] Commits are split per component (schema, portal/platform, kamusers, agis, ..)
- [x] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

**Review by ironArt3mis — CHANGES_REQUESTED** [review]
Functional review required

File: web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  const changeColumnMappingHandler = (event: SelectChangeEvent) => {
+    const { name, value } = event.target;
+    const newVal = [...columns];
+    newVal[parseInt(name, 10)] = value;
+    setColumns(newVal);
+  };
+
+  const changeIgnoreFirstLineHandler = (
+    event: React.ChangeEvent<HTMLInputElement>,
```
Comment (rbatistadev) [inline web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx:41]: Variable not used, please remove.

File: web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  ) => {
+    setIgnoreFirstLine(checked);
+  };
+
+  return (
+    <>
+      <StyledTable size='medium' sx={{ overflowY: 'auto' }}>
+        <TableHead>
+          <TableRow>
+            {csv[0].map((row, key) => {
```
Comment (rbatistadev) [inline web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx:52]: ignore row here if you are not using it
```suggestion
            {csv[0].map((_, key) => {
```

File: web/portal/client/src/entities/HolidayDate/Action/Import.tsx [hunk 3]
```diff
@@ -41,47 +51,126 @@ const fileToBlob = (file: File): Blob => {
...
+    entityService,
+    fixedValues: {},
+    filterValues: {},
+    filterBy: undefined,
+    initialValues: row,
+    validator: (values) => values as EntityValidatorResponse,
+    onSubmit: async () => {
+      /* noop */
+    },
+  });
```
Comment (rbatistadev) [inline web/portal/client/src/entities/HolidayDate/Action/Import.tsx:69]: You are not using formik here

File: web/portal/client/src/entities/HolidayDate/Action/Import.tsx [hunk 4]
```diff
@@ -20,11 +22,19 @@ import {
...
+import FileUpload from 'components/FileUpload';
+import Papa from 'papaparse';
+import { useEffect, useState } from 'react';
 import { useStoreActions, useStoreState } from 'store';
 
-import FileUpload from '../../../components/FileUpload';
 import HolidayDate from '../HolidayDate';
+import ImportHolidayDatesMappingTable from './ImportHolidayDates';
+
+type importerResults = {
```
Comment (rbatistadev) [inline web/portal/client/src/entities/HolidayDate/Action/Import.tsx:33]:
```suggestion
type ImporterResults = {
```

File: web/portal/client/src/entities/HolidayDate/Action/Import.tsx [hunk 5]
```diff
@@ -41,47 +51,126 @@ const fileToBlob = (file: File): Blob => {
...
+      })
+    );
+
+    const parseImporterResults = (data: importerResults) => {
+      return {
+        errorMsg: data.errorMsg,
+        success: data.success,
+        failed: data.failed,
+      } as importerResults;
+    };
```
Comment (rbatistadev) [inline web/portal/client/src/entities/HolidayDate/Action/Import.tsx:153]:
```suggestion
    const parseImporterResults = (data: importerResults): ImporterResults => {
      return {
        errorMsg: data.errorMsg,
        success: data.success,
        failed: data.failed,
      };
    };
```

File: library/Ivoz/Provider/Application/Service/HolidayDate/SyncFromCsv.php [hunk 6]
```diff
@@ -17,56 +21,107 @@ public function __construct(
...
+            $this->entityTools->persist($holidayDate);
+        } catch (\Exception $e) {
+            $isSafeException = ($e instanceof \DomainException);
+
+            $errorMsg = $isSafeException
+                ? $e->getMessage()
+                : 'Unknown error';
+
+            $this->addError($line + 1, $errorMsg);
         }
```
Comment (rbatistadev) [inline library/Ivoz/Provider/Application/Service/HolidayDate/SyncFromCsv.php:117]:
```suggestion
        } catch (\DomainException $e) {
            $this->addError($line + 1, $e->getMessage());
        }catch (\Exception $e) {
            $this->addError($line + 1, 'Unknown error');
        }
```

File: web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx (hunk 1 above)
Comment (danigargar) [reply web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx:41]: https://github.com/irontec/ivozprovider/blob/fec020256e746d8672e2ffdbf7f468bce8337c07/web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx#L100

Am I missing something?

Comment (rbatistadev) [reply web/portal/client/src/entities/HolidayDate/Action/ImportHolidayDates.tsx:41]: If you are not using the variable but is needed for any other reason you can rewrite it like "_"

**Review by rbatistadev — APPROVED** [review]

File: web/portal/client/src/entities/HolidayDate/Action/Import.tsx [hunk 7]
```diff
@@ -113,20 +200,70 @@ const Import: ActionFunctionComponent = (props: MultiSelectActionItemProps) => {
         onClose={handleClose}
         aria-labelledby='alert-dialog-title'
         aria-describedby='alert-dialog-description'
+        className={step === 1 ? '' : 'width'}
       >
         <Box>
           <DialogTitle id='alert-dialog-title'>
-            {_('Import Holiday Dates')}
+            {_('Import Holiday dates')}
```
Comment (Kaian) [inline web/portal/client/src/entities/HolidayDate/Action/Import.tsx:207]: If you change this capital letter, be sure to update all the texts with the same label (line 184 and 188 of this same file) to minimize the number of translatable strings. Choose one (dates or Dates) and update all occurrences.

**Review by Kaian — CHANGES_REQUESTED** [review]

File: web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php [hunk 8]
```diff
@@ -42,27 +45,42 @@ public function __invoke(Request $request)
...
 
         $calendarId = $request->request->get('calendar');
+
+        /** @var CalendarInterface|null $calendar */
+        $calendar = $this->calendarRepository->find($calendarId);
+        if (is_null($calendar)) {
+            throw new NotFoundHttpException('Calendar not found');
+        }
+
+        $calendarBelongsToCompany = $calendar->getCompany()->getId() === $company->getId();
```
Comment (Kaian) [inline web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php:55]: What about a new repository method findCalendarByCompanyId($companyId, $calendarId), not worth it?

Comment (Kaian) [reply web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php:55]: Or maybe findCompanyCalendar($companyId, $calendarId) following the nomenclature found here: https://github.com/irontec/ivozprovider/blob/6bcc2584e430b2773ac4d2030b927e6dd8828393/library/Ivoz/Provider/Domain/Model/Extension/ExtensionRepository.php#L19

File: web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php [hunk 9]
```diff
@@ -26,7 +29,7 @@ public function __construct(
      */
     public function __invoke(Request $request)
     {
-        $token = $this->tokenStorage->getToken();
+            $token = $this->tokenStorage->getToken();
```
Comment (Kaian) [inline web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php:32]: Remove extra space

File: web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php [hunk 10]
```diff
@@ -6,8 +6,10 @@
 use Ivoz\Core\Domain\DataTransferObjectInterface;
 use Ivoz\Provider\Application\Service\HolidayDate\SyncFromCsv;
 use Ivoz\Provider\Domain\Model\Administrator\AdministratorInterface;
+use Ivoz\Provider\Domain\Model\Calendar\CalendarInterface;
+use Ivoz\Provider\Domain\Model\Calendar\CalendarRepository;
 use Model\HolidaysMassImport;
-use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Request;
```
Comment (Kaian) [inline web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php:12]: Remove extra space. I though test-codestyle will cover this.

File: web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php [hunk 11]
```diff
@@ -42,27 +45,37 @@ public function __invoke(Request $request)
         }
 
         $calendarId = $request->request->get('calendar');
+        $companyId = $company->getId() ?? -1;
```
Comment (Kaian) [inline web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php:48]: I don't see the point on continuing without a valid companyId, check in the above condition (line 43) if company has value and it is valid

        $company = $user->getCompany();
        if (!$company || !$company->getId()) {
            throw new NotFoundHttpException('Company not found');
        }

or 

        $company = $user->getCompany();
        if (!$company) {
            throw new NotFoundHttpException('Company not found');
        }

        $companyId = $company->getId();
        if (!$companyId) {
            throw new NotFoundHttpException('Company not valid');
        }

**Review by Kaian — CHANGES_REQUESTED** [review]

File: web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php (hunk 11 above)
Comment (danigargar) [reply web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php:47]: Execution is not reaching there if there is no valid company (lines 42-44).

It's a little trick to keep phpstan and psalm relaxed without error supresion. There's no chance that any entity's getId() returns null except if it's new. 

Other options are not that effective:
````
if (!$company || is_null($company->getId)) {
      throw new NotFoundHttpException('Company not found');
}
```
This don't prevents psalm complaining about a possible null.

**Review by Kaian — APPROVED** [review]

**Pull request merged**